CMS: Make Content Overview mobile friendly

DirkDevelopmentFreelance opened the pull request.

DirkDevelopmentFreelance commented:
This does not look nice, so I decided to rather move translations to it's own row on mobile
![image](https://github.com/user-attachments/assets/a9bf9f2b-3420-40f5-80f0-cb9a9e6fddcb)
Updated:
![image](https://github.com/user-attachments/assets/d1a55a4a-f818-4175-9e11-1b0874ceda05)

ivanslabbert commented:
- [ ] Long titles not truncating / wrapping at correct place
![image](https://github.com/user-attachments/assets/85fcf69c-ca3d-4659-8846-d11f6fe79205)

- [ ] Move updated date time to middle row
![image](https://github.com/user-attachments/assets/f3d7a2ac-c584-4780-9f44-43e7f196b4df)

- [ ] Fix double scrolling issue
- [ ] Left-align publish and expiry dates
- [ ] Include selected tags and groups in filter bar card
![image](https://github.com/user-attachments/assets/54bc5d71-8e8f-49c4-a91c-e58fec3629c2)

- [ ] Overflow not set to "hidden" on the search bar
![image](https://github.com/user-attachments/assets/96f6af7a-819b-43b8-b424-1d7dca1375a0)

- [ ] Move navigation to footer slot
![image](https://github.com/user-attachments/assets/4755eed0-b5c7-4dad-92d5-c091d3308748)

- [ ] In mobile, hide the page size selector (dropdown), and only show for 3 seconds when the user press a scroll or page selection button
![image](https://github.com/user-attachments/assets/ca2ad6d8-f902-495a-acf2-5ad212b035eb)

- [ ] Remove publish and expiry date from cards (both mobile and desktop)
![image](https://github.com/user-attachments/assets/649658ef-b7c1-4ef0-b0a1-693428d650c6)

DirkDevelopmentFreelance commented:
Progress made today (2025/05/26)
1. Moved the Create buttons and titles on overviews into the `TopBar`
2. Created a new component called `TopBarActions` to handle the create buttons and titles dynamically
3. Fixed tests
4. Made the LPaginator  more mobile friendly

Todo:
Make changes requested from standup (2025/05/27):
      - [ ]
      - [ ]
      - [ ]
- [ ] Fix backlink in `EditContent.vue`
- [ ] Fix failing tests
- [ ] Move tests to relevant places if necessary
- [ ] Create tests if relevant and needed

DirkDevelopmentFreelance commented:
It won't be possible to disable root app scrolling as too many components rely on it right now, I think it can be fixed once all pages are converted to be mobile friendly

DirkDevelopmentFreelance commented on cms/src/components/BasePage.spec.ts:
```diff
@@ -13,7 +28,7 @@ describe("BasePage", () => {
     });
 
-    it("renders the back link", async () => {
+    it.skip("renders the back link", async () => {
```
> [!NOTE]
> Maybe we can create a ticket for this test, as the back link is rendering, it is weird routing issues going on in the testing env, but then this can get merged perhaps

ivanslabbert replied:
Please create a separate ticket for this?

DirkDevelopmentFreelance commented on cms/src/components/content/ContentOverview/ContentOverview.spec.ts:
```diff
@@ -72,7 +85,7 @@ describe("ContentOverview.vue", () => {
     });
 
-    it("should show edit button with correct router link and icon", async () => {
+    it.skip("should show edit button with correct router link and icon", async () => {
```
> [!NOTE]
> The edit button has been removed, so this test can perhaps also be removed

DirkDevelopmentFreelance commented on cms/src/pages/SettingsPage.spec.ts:
```diff
@@ -1,62 +1,120 @@
+        await db.docs.clear();
+        await db.localChanges.clear();
+    });
     it("purges the local database when connected", async () => {
```
> [!NOTE]
> I changed this test, as this was only testing UI reactions in my opinion, it was not testing the actual functionality, and this resulted in the test failing and the fix was a massive workaround, this was the cleaner approach.

ivanslabbert replied:
Agree with the approach, but need a separate test to check that the notification still shows.

DirkDevelopmentFreelance commented:
> [!NOTE]
>  A visual bug I found on 25/06/02 --> Just a z-index issue, is very easy to fix
![image](https://github.com/user-attachments/assets/f8fda4a4-c4d1-42b5-98bc-6b60c90223c5)

DirkDevelopmentFreelance commented:
Notes from stand up 09:36 review
More bottom padding on desktop for paginator
Bleed filter options to sides
Move last updated to top (Not centered)
LPaginator paging size must stay open before selected
Try gap-2 on card seperation
Look at visual bug for LPaginator

DirkDevelopmentFreelance commented:
>  Move last updated to top (Not centered) 
![image](https://github.com/user-attachments/assets/a1fcb518-8ca8-4f52-8e40-122b9cf4c164)
Already at top, would have to remove padding to make this happen

ivanslabbert commented on cms/src/components/content/ContentOverview/ContentOverview.spec.ts:
```diff
@@ -103,7 +116,7 @@ describe("ContentOverview.vue", () => {
     });
 
-    it("should show view icon with correct router link if no edit permission", async () => {
+    it.skip("should show view icon with correct router link if no edit permission", async () => {
```
Can be removed

ivanslabbert commented on cms/src/components/content/ContentOverview/ContentOverview.spec.ts:
```diff
@@ -443,4 +428,36 @@ describe("ContentOverview.vue", () => {
+                expect(linkProps.params?.id).toBe("new");
+            });
+        },
+        { timeout: 10000 },
```
Is this needed?

ivanslabbert reviewed (approved):
Approved with some comments
